gamemodes/terrortown/gamemode/server/sv_player.lua

```diff
@@ -512,6 +512,72 @@ local function SpecUseKey(ply, ent)
     end
 end
 
+local function EntityContinuousUse(ent, ply)
+    ---
+    -- Enable addons to allow handling PlayerUse
+    -- Otherwise default to old IsTerror Check
+    -- @realm server
+    -- stylua: ignore
+    if hook.Run("PlayerUse", ply, ent, ply:IsTerror()) then
+        ent:Use(ply, ply)
+    end
+
+    if ply:IsSpec() then
+        SpecUseKey(ply, ent)
+
+        return
+    end
+
+    if not ply:IsTerror() then
+        return
+    end
+
+    if ent.CanUseKey and ent.UseOverride then
+        local phys = ent:GetPhysicsObject()
+
+        if not IsValid(phys) or phys:HasGameFlag(FVPHYSICS_PLAYER_HELD) then
+            return
+        end
+
+        ent:UseOverride(ply)
+    elseif ent.player_ragdoll then
+        CORPSE.ShowSearch(ply, ent, ply:KeyDown(IN_WALK) or ply:KeyDownLast(IN_WALK))
+
+        return
+    elseif ent:IsWeapon() then
+        ply:SafePickupWeapon(ent, false, true, true, nil)
+
+        return
+    end
+
+    -- if it is a SENT, this will always return true
+    -- if it is a map entity, the flag will be checked
+    if not ent:IsUsableEntity(FCAP_CONTINUOUS_USE) then
+        return
+    end
+
+    -- make sure it is called 10 times per second
+    timer.Simple(0.1, function()
+        if not IsValid(ent) or not IsValid(ply) then
+            return
+        end
+
+        -- make sure the use key is still pressed
+        if not ply:KeyDown(IN_USE) then
+            return
+        end
+
+        -- make sure the entity is still in a good position
+        local distance = ply:GetShootPos():Distance(ent:GetPos())
+
+        if distance > 100 + ent:BoundingRadius() then
```

Comment:
This is defined in two positions, therefore changing the hardcoded 100 could be a hassle.
Also this local function does not repeat for buttons.

I wonder if the continuous use function shouldnt be combined with the rest of the code inside TTT2PlayerUseEntity.

Reply:
> This is defined in two positions, therefore changing the hardcoded 100 could be a hassle.

I'm not sure what you mean. Where is this defined again?

> I wonder if the continuous use function shouldnt be combined with the rest of the code inside TTT2PlayerUseEntity.

We could. But we didn't do it before. This only fixes the issue that that change introduced. I'm open to both though

Reply:
@ZenBre4ker could you please elaborate here? I don't know what you expect me to change, sorry

Reply:
i'm just going to hand-waive this so we can go forward with this

Reply:
I can't merge though as long as @ZenBre4ker has requested changes open